xagon-react-app/src/components/GameComponent/inputManager/FlipGesture.ts

```diff
@@ -0,0 +1,180 @@
+import {
+  AbstractMesh,
+  PointerInfo,
+  Scene,
+  Nullable,
+  StandardMaterial,
+  Vector3,
+} from '@babylonjs/core';
+import { getAssetMesh } from 'utils/scene';
+import Gesture from './Gesture';
+
+interface GestureContext {
+  scene: Scene;
+  triangleMesh: AbstractMesh;
+  scalingRatio: number;
+}
+
+class FlipGesture extends Gesture {
+  private context: GestureContext;
+
+  private firstTriangle: Nullable<AbstractMesh>;
+
+  private secondTriangle: Nullable<AbstractMesh>;
+
+  public constructor(context: GestureContext) {
+    super();
+    this.context = context;
+    this.firstTriangle = null;
+    this.secondTriangle = null;
+  }
+
+  public computeObjSpaceData(assetMesh: AbstractMesh):
+    | {
+        vertices: Vector3[];
+        edges: Vector3[];
+      }
+    | undefined {
+    if (assetMesh && assetMesh.skeleton) {
+      const objSpaceVertices = assetMesh.skeleton.bones.map((bone) =>
+        bone
+          .getDirection(this.context.triangleMesh.up)
+          .scale(this.context.scalingRatio),
+      );
+
+      const edges = objSpaceVertices.map((v, i) =>
+        v.subtract(objSpaceVertices[(i + 1) % 3]),
+      );
+
+      return { vertices: objSpaceVertices, edges };
+    }
+    // eslint-disable-next-line no-console
+    console.assert(typeof assetMesh === 'object', 'Asset not found');
+    return undefined;
+  }
+
+  public onDown(pointerInfo: PointerInfo): void {
+    const mesh = pointerInfo?.pickInfo?.hit && pointerInfo.pickInfo.pickedMesh;
+    if (mesh) {
+      const assetMesh = getAssetMesh({
+        scene: this.context.scene,
+        triangleMesh: mesh,
+      });
+      if (assetMesh) {
+        const data = this.computeObjSpaceData(assetMesh);
+        if (data) {
+          assetMesh.metadata.vertices = data.vertices;
+          assetMesh.metadata.edges = data.edges;
+
+          this.firstTriangle = assetMesh;
+        }
+      }
+    }
+  }
+
+  public onMove(): void {
+    if (this.firstTriangle) {
+      const firstTriangle = this.firstTriangle.metadata.triangle;
```

Comment:
tipo qui, this.firstTriangle e firstTriangle sono 2 cose diverse benche' abbiano lo stesso nome